cephfs/admin/subvolume.go

```diff
@@ -201,6 +201,7 @@ const (
 type SubVolumeInfo struct {
 	Type          string    `json:"type"`
 	Path          string    `json:"path"`
+	State         string    `json:"state"`
```

Comment:
This should probably be an "enumerated" string with a type the way Feature is for the Features field.
Something like:
```go
type SubVolumeState string
const (
    StateUnset =       SubVolumeState("")
    StateInitalizing = SubVolumeState("init")
    StatePending =     SubVolumeState("pending")
    StateInProgress =  SubVolumeState("in-progress")
    StateFailed =      SubVolumeState("failed")
    StateComplete =    SubVolumeState("complete")
    StateCanceled =    SubVolumeState("canceled")
    StateRetained =    SubVolumeState("snapshot-retained")
)
```
Note: I got the states from https://github.com/ceph/ceph/blob/8bb2d5ecc900bbfcfae3dd7cc53a37b1bbabca45/src/pybind/mgr/volumes/fs/operations/versions/subvolume_attrs.py#L22-L28

Reply:
Yeah, I saw that but unlike Features this can have just one value (same as Type) that's why I didn't do it.
But now on second thoughts, it would be good to have as it will make the output strict.